### DISCUSSION TITLE
nOOb question (concerning SSL requirement)

### DISCUSSION THREAD

**Discussion by Nothing2Do** [Ylianst/MeshCentral#4418] [Q&A] [answered]
Some time ago (1 year or so), I've read that there are a requirement on SSL (the version to install, I think but I've forgot exactly which).
And now, my problem concerning OTP seem to be related. I've search for a long time but I'm unable to get it. Do you know a openssl requirement ? output of "$ openssl version" is "OpenSSL 3.0.4 21 Jun 2022 (Library: OpenSSL 3.0.4 21 Jun 2022)" ...

**Comment by Ylianst** [comment]
Not sure exactly what your question is, but MeshCentral makes use of TLS 1.2 and TLS 1.3 on the main HTTPS port which is normally set to 443. MeshCentral uses the OpenSSL library that is built into NodeJS to perform TLS operations, so you should keep your NodeJS version up to date to make sure it have the latest security fixes.

As for as the MeshCentral agent is concerned, it's compiled directly with OpenSSL and uses the 1.1.1 branch. We may make the switch to OpenSSL 3.0 at some point. Note that both OpenSSL 1.1.1 and 3.0 branches are currently supported and so, using one or the other is ok, as long as TLS 1.2 or better is used.

Hope that helps.

**Comment by Nothing2Do** [comment]
I'll try to "clarify", I've use meshcentral for a long time and since, at least 2 years (I was active in the github page back then). And now, every time I register a second authentification factor (a backup code, a push notification, an OTP "google authenticator", a mail OTP, a bakup code, a yubikey, or whatever ...), I can't log'in anymore. Every OTP code I enter, seem wrong.
I have to stop MC, run "node ~/node_modules/meshcentral --cert $SITE --resetaccount $USER --domain $SITE --pass 'newpass'" and relaunch MC to log'in again.

**Reply by si458** [reply]
are you running the latest version of meshcentral 1.1.19? and using the latest NODEJS 20?
normally when OTP isnt working its because the TIME of the meshcentral server is INCORRECT at the time you setup the OTP?
and im aware the push notification doesnt work but will hopefully start working again in the next release!

**Reply by Nothing2Do** [reply]
yes, the latest ("freebox@plex:~$ node -v / v20.11.0" and meshcentral is uploaded). 
it's not really the OTP code that's wrong, the button I click on for my yubikey (or mail, or push, ...) redirect me on the first page (with the user and password to enter) without sending anyhing. but my memory play with me, I think you've already helped me ... by desperation : "$ date && ssh plex date
Sat Jan 27 10:33:22 AM CET 2024
Sat 27 Jan 2024 10:33:23 AM CET"

**Reply by si458** [reply]
That's weird? If u setup otp (that's the most basic 2fa method) and when u login, does it ask for the 2fa code? Or is it being redirected?

**Reply by Nothing2Do** [reply]
it ask for otp code, but it always redirect me to the front page when I validate my OTP code (double checked).

**Reply by si458** [reply]
Did it it come up with any errors on the login page when it redirected?
Because this to me sounds like a proxy redirect issue?
Can u share ur config.json (hide secrets etc) and do u use any proxy in front of meshcentral?

**Reply by Nothing2Do** [reply]
no errors, only an empty front page.  and no proxy (other than my ISP box ...)
[plex.txt](https://github.com/Ylianst/MeshCentral/files/14071862/plex.txt)

**Reply by si458** [reply]
You have 2 dns names listed in the cert bit at the top,
I believe you can only use a single dns there (ur main dns name)
Then any subdomains, get listed under there own domains sections (as u have already) and a dns entry listed (again like u have) 
And u then put the multiple domains in the letsencrypt name section

Give that a try!?

**Reply by Nothing2Do** [reply]
I've tryed, no difference.

**Reply by si458** [reply]
have u checked the console output of meshcentral itself on your server?
do you get any errors there? generally a white page means an error server side,
if you check the web browser console network tab, (i use firefox then right click the login page and click inspect, then network),
when you get the white page is the page an error 500? or 404? or anything different?
do you have any customizations in meshcentral-web ?

i think you might have to start with a fresh install (stop meshcentral, remove node_modules completely, npm install meshcentral, start meshcentral)

**Reply by Nothing2Do** [reply]
I've started from scratch, with the config.json I gave you and it still don't work ... and I have no white page, it redirect me on the login page.

**Reply by si458** [reply]
you have `minify` set as true, try changing that to `false` and restart meshcentral, then try in `incognito/private window` ?

EDIT: also try removing `maxinvalid2fa` completely, then restart, and same
also try removing `cookieEncoding` too then restarting...

it must b e something it doesnt like with your config...

**Reply by si458** [reply]
try this config - [tryme.txt](https://github.com/Ylianst/MeshCentral/files/14072080/tryme.txt)

**Reply by Nothing2Do** [reply]
I've put my password into your file as you send it (no other modification) and it still don't work :-( (you have removed compression stuff).

**Reply by si458** [reply]
yes i removed all advanced stuff and set the very basic for you,
but if its STILL redirecting back to the login page with NO ERROR showing at all, not even 'Invalid token, try again.' being shown?
then i think it MUST be something with your machine/setup, please open a BUG report on issues, and follow template and include ALL DETAILS please as that is bloody weird!?

**Reply by Nothing2Do** [reply]
I remember that it has something to do with openssl 1.1.1n ("openssl version"), but I can't remember howto use an alternative (modulo, my faulty memory).

**Reply by si458** [reply]
what OS are you running? and what openssl version?
im using Ubuntu 22.04 and OpenSSL 3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)

**Reply by Nothing2Do** [reply]
the meshcentral server is into "plex" (a virtual box inside my local ISP modem/router)
$ ssh plex cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
and
$ ssh plex openssl version -v
OpenSSL 1.1.1n  15 Mar 2022
I wouldlike to use openssl 3, but I can't find a way to install it on debian without recompiling it ...

**Reply by si458** [reply]
I will investigate when I get chance

**Reply by Nothing2Do** [reply]
If I can help, I'll do my best ...

**Reply by Nothing2Do** [reply, marked as answer]
news from the front, so long without a correct mesh instance ... I decided to recompile openssl from source. Now "freebox@plex:~$ openssl -version
OpenSSL 3.2.0 23 Nov 2023 (Library: OpenSSL 3.2.0 23 Nov 2023)", and it recognize my yubikey,I can log with it. my other OTP work too.  I'm happy (but i'm sad that I don't know why it has stopped working)

**Reply by silversword411** [reply]
...and in the deprecation of all TLS < 1.2 dept....

**Reply by Nothing2Do** [reply]
and life goes on, forget the past and your old habit/software/way as always ...
